node-wasm/src/client.rs

```diff
@@ -409,3 +409,145 @@ impl WasmNodeConfig {
         })
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::time::Duration;
+
+    use super::*;
+    use crate::worker::NodeWorker;
+    use celestia_rpc::{prelude::*, Client};
+    use celestia_types::ExtendedHeader;
+    use libp2p::{multiaddr::Protocol, Multiaddr};
+    use rexie::Rexie;
+    use serde_wasm_bindgen::from_value;
+    use tracing_subscriber::filter::LevelFilter;
+    use tracing_subscriber::{
+        fmt::time::UtcTime, layer::SubscriberExt, util::SubscriberInitExt, Layer,
+    };
+    use tracing_web::MakeConsoleWriter;
+    use wasm_bindgen_futures::spawn_local;
+    use wasm_bindgen_test::wasm_bindgen_test;
+    use web_sys::MessageChannel;
+
+    const WS_URL: &str = "ws://localhost:36658";
+
+    pub async fn fetch_bridge_webtransport_multiaddr(client: &Client) -> Multiaddr {
+        let bridge_info = client.p2p_info().await.unwrap();
+
+        let mut ma = bridge_info
+            .addrs
+            .into_iter()
+            .find(|ma| {
+                let not_localhost = !ma
+                    .iter()
+                    .any(|prot| prot == Protocol::Ip4("127.0.0.1".parse().unwrap()));
+                let webtransport = ma
+                    .protocol_stack()
+                    .any(|protocol| protocol == "webtransport");
+                not_localhost && webtransport
+            })
+            .expect("Bridge doesn't listen on webtransport");
+
+        if !ma.protocol_stack().any(|protocol| protocol == "p2p") {
+            ma.push(Protocol::P2p(bridge_info.id.into()))
+        }
+
+        ma
+    }
+
+    async fn remove_database() -> rexie::Result<()> {
+        Rexie::delete("private").await?;
+        Rexie::delete("private-blockstore").await?;
+        Ok(())
+    }
+
+    fn setup_logs() {
+        let fmt_layer = tracing_subscriber::fmt::layer()
+            .with_ansi(false)
+            .with_timer(UtcTime::rfc_3339()) // std::time is not available in browsers
+            .with_writer(MakeConsoleWriter) // write events to the console
+            .with_filter(LevelFilter::DEBUG); // TODO: allow customizing the log level
+        tracing_subscriber::registry().with(fmt_layer).init();
+    }
```

Comment:
can we re-use the setup logging from utils?

Reply:
This was leftover from debugging, we don't setup logging on native tests either. Removed :+1: